src/f4/app_cantest/source/test_time.c

```diff
@@ -16,42 +16,42 @@ void cmd_time(BaseSequentialStream *chp, int argc, char *argv[])
     if (argc < 1) {
         goto time_usage;
     }
-    if (!strcmp(argv[0], "unix")) {
-        if (!strcmp(argv[1], "get")) {
+    if (argv[0] != NULL && !strcmp(argv[0], "unix")) {
+        if (argv[1] != NULL && !strcmp(argv[1], "get")) {
             time_t unix_time = rtcGetTimeUnix(&msec);
             char *timestr = ctime(&unix_time);
             chprintf(chp, "UNIX Time: %d\r\n"
                           "Date:      %s\r\n",
                           unix_time, timestr);
-        } else if (!strcmp(argv[1], "set") && argc > 2) {
+        } else if (argv[1] != NULL && !strcmp(argv[1], "set") && argc > 2) {
```

Comment:
In this (and similar) situations this actually creates a new subtle bug. You're checking if the value at the location is zero'ed out- which sometimes will be true- sometimes wont. This also assumes that this is in-bounds.

Me thinks it would be better to check against `argc` and ensure it is greater than or equal to the current spot you are trying to deference.

For example: `argv[0]` is technically always a valid pointer, but it's also a de-reference operation that's not guaranteed to work if the array was never allocated, or not allocated with the size you assume. This code would break if we simply passed no args to this tool.

Reply:
I've updated the description for now, will make it more detailed ASAP. Will also make this conditional/similar more sturdy, likely today or tomorrow.
Thanks!